pupil-spa/src/app/login-success/login-success.component.ts

```diff
@@ -25,9 +26,23 @@ export class LoginSuccessComponent implements OnInit, AfterViewInit, OnDestroy {
               private questionService: QuestionService,
               private speechService: SpeechService,
               private appUsageService: AppUsageService,
+              private userService: UserService,
               private elRef: ElementRef) {
     const pupilData = storageService.getItem('pupil');
     const schoolData = storageService.getItem('school');
+
+    // if the user comes back to this component after his personal data
+    // was deleted (direct access through link or the back button in browser)
+    // redirect him according to his status - logged in or not
+    if (pupilData === null || schoolData === null) {
```

Comment:
@danwilliams would you mind changing this nested if statement to a guard clause? Personally I believe nested if statements decrease code readability even though yours is only 1 level down.
See this link: [https://refactoring.guru/replace-nested-conditional-with-guard-clauses](https://refactoring.guru/replace-nested-conditional-with-guard-clauses)